src/incoming-message-stream.ts

```diff
@@ -82,7 +82,7 @@ class IncomingMessageStream extends Transform {
           return;
         }
       } else {
-        break;
+        return callback(new Error('Unable to process incoming buffer'));
```

Comment:
I think we should return a `ConnectionError` here to be a bit more specific, instead of a generic `Error`.